### PR TITLE
libxml2 2.9.13

### PR DESCRIPTION
**Pull request by fanninpm**
Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by fanninpm** [comment]
@carlocab tests appear to pass now, but I'm getting hung up when I try to upgrade to `python@3.10`.

**Comment by cho-m** [comment]
Not sure if it should have caused failure, but probably should clean up the function's usage of deprecated `on_macos/on_linux`:
```rb
  def sdk_include
    on_macos do
      return MacOS.sdk_path/"usr/include"
    end
    on_linux do
      return HOMEBREW_PREFIX/"include"
    end
  end
```

Either a ternary:
```rb
  def sdk_include
    return OS.mac? ? MacOS.sdk_path/"usr/include" : HOMEBREW_PREFIX/"include"
  end
```

Or just an `if OS.mac? ... else ... `

Could also just run directly inside `install` block rather than using function.

File: Formula/libxml2.rb [hunk 1]
```diff
@@ -41,7 +42,7 @@ class Libxml2 < Formula
 
   keg_only :provided_by_macos
 
-  depends_on "python@3.9" => [:build, :test]
+  depends_on "python@3.10" => [:build, :test]
```
Comment (carlocab) [inline Formula/libxml2.rb:45]: If we want to change this, we need to check that the following formulae don't make use of `libxml2`'s Python bindings:
```
anjuta
at-spi2-atk
gdal
gtk-doc
gupnp
itstool
yelp-tools
```

Comment (fanninpm) [reply Formula/libxml2.rb:45]: `anjuta` makes use of those bindings. 😕

Comment (carlocab) [reply Formula/libxml2.rb]: Migrating `anjuta` will be a whole thing (it has `glib` in its dep tree, and that connects it to a pretty large fragment of homebrew-core dependency graph), so I suggest we stick with `python@3.9` for now.

When you make changes here can you squash your commits together? Thanks!

**Comment by cho-m** [comment]
Interesting that it complains here and also complains about the opposite pattern. Seems like checks aren't getting applies consistently somewhere in Homebrew.

May just move the logic from the function into the `install` section 
```rb
cd "python" do
  sdk_include = OS.mac? ? MacOS.sdk_path/"usr/include" : HOMEBREW_PREFIX/"include"
  ...
```

**Comment by carlocab** [comment]
Most of the failures on Linux look like the usual ones. However, there are some that might need investigating: `gromacs` and `libxslt`. `libxslt` in particular looks like a regression.

I assume `v8` is just from the linker getting confused about Homebrew `icu4c` vs the bundled one.

CC @Homebrew/linux

**Comment by fanninpm** [comment]
> Have you tested `libxslt` 1.1.35 with this version of `libxml2` locally?

`brew install -s --include-test libxml2 && brew install -s --include-test libxslt && brew test libxslt` seems to work on my machine (macOS 12.1.2 on M1). Is there a way to test the Linux portion via Docker?

**Comment by carlocab** [comment]
> Is there a way to test the Linux portion via Docker?

Yes. See: https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md#homebrew-docker-container

**Review by carlocab — DISMISSED** [review]
If `libxslt` does not fail on Linux I think this should be good to merge.

**Comment by cho-m** [comment]
We are going to have to rebuild `libxslt` due to common issue where our `pkgconfig` and similar files all embed the real versioned Cellar path, e.g.
```console
$ xslt-config --cflags --libs
-I/home/linuxbrew/.linuxbrew/Cellar/libxml2/2.9.12/include/libxml2 -I/home/linuxbrew/.linuxbrew/Cellar/libxslt/1.1.35/include -L/home/linuxbrew/.linuxbrew/Cellar/libxslt/1.1.35/lib -L/home/linuxbrew/.linuxbrew/Cellar/libxml2/2.9.12/lib -lxslt -lxml2 -lz -ldl -lm
```

May want to inreplace `xslt-config` to use `Formula["libxml2"].opt_include` rather than `Formula["libxml2"].include`

**Comment by cho-m** [comment]
Might as well list Linux failures to resolve outside this PR:
- [x] `amdatu-bootstrap` - fixed in #96529
- [ ] `giza`
- [ ] `gromacs`
- [ ] `gst-libav` - working on in #95493
- [x] `javarepl` - deprecated in #96530
- [ ] `jetty`
- [ ] `libxslt`
- [ ] `mrbayes`
- [ ] `redpen`
- [ ] `solr@7.7`
- [ ] `streamlink`
- [ ] `xmrig`

**Comment by cho-m** [comment]
I would recommend updating `libxslt` to avoid this in future. Something like:
```diff
   def install
+    libxml2 = Formula["libxml2"]
     system "autoreconf", "--force", "--install", "--verbose" if build.head?
     system "./configure", "--disable-dependency-tracking",
                           "--disable-silent-rules",
                           "--prefix=#{prefix}",
                           "--without-python",
                           "--with-crypto",
-                          "--with-libxml-prefix=#{Formula["libxml2"].opt_prefix}"
+                          "--with-libxml-prefix=#{libxml2.opt_prefix}"
     system "make"
     system "make", "install"
+    inreplace bin/"xslt-config", libxml2.prefix.realpath, libxml2.opt_prefix
   end
```

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**